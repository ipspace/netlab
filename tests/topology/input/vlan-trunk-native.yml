defaults.device: linux

vlans:
  red:
    ospf.cost: 10
  blue:
  green:
nodes:
  s1:
    device: iosv
    module: [ vlan,ospf ]
  s2:
    device: eos
    module: [ vlan,ospf ]
  h1:
  h2:
  h3:
  h4:
  h5:

links:
- s1:
    vlan.access: red
  h1:
- s1:
    vlan.access: blue
  h2:
- s2:
    vlan.access: red
  h3:
- s2:
    vlan.access: blue
  h4:
- s2:
    vlan.access: green
  h5:
- s1:
    vlan.trunk: [ red, blue ]
  s2:
    vlan.trunk: [ red, blue, green ]
    vlan.native: green
  h1:
- s1:
  s2:
  vlan:
    trunk: [ red, blue, green ]
    native: green
