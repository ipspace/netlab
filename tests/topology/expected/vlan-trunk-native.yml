input:
- topology/input/vlan-trunk-native.yml
- package:topology-defaults.yml
links:
- bridge: input_1
  gateway:
    ipv4: 172.16.0.1/24
  interfaces:
  - _vlan_mode: irb
    ifindex: 1
    ifname: GigabitEthernet0/1
    ipv4: 172.16.0.1/24
    node: s1
    vlan:
      access: red
  - ifindex: 1
    ifname: eth1
    ipv4: 172.16.0.3/24
    node: h1
  linkindex: 1
  node_count: 2
  ospf:
    cost: 10
  prefix:
    allocation: id_based
    ipv4: 172.16.0.0/24
  type: lan
- bridge: input_2
  gateway:
    ipv4: 172.16.1.1/24
  interfaces:
  - _vlan_mode: irb
    ifindex: 2
    ifname: GigabitEthernet0/2
    ipv4: 172.16.1.1/24
    node: s1
    vlan:
      access: blue
  - ifindex: 1
    ifname: eth1
    ipv4: 172.16.1.4/24
    node: h2
  linkindex: 2
  node_count: 2
  prefix:
    allocation: id_based
    ipv4: 172.16.1.0/24
  type: lan
- bridge: input_3
  gateway:
    ipv4: 172.16.0.2/24
  interfaces:
  - _vlan_mode: irb
    ifindex: 1
    ifname: Ethernet1
    ipv4: 172.16.0.2/24
    node: s2
    vlan:
      access: red
  - ifindex: 1
    ifname: eth1
    ipv4: 172.16.0.5/24
    node: h3
  linkindex: 3
  node_count: 2
  ospf:
    cost: 10
  prefix:
    allocation: id_based
    ipv4: 172.16.0.0/24
  type: lan
- bridge: input_4
  gateway:
    ipv4: 172.16.1.2/24
  interfaces:
  - _vlan_mode: irb
    ifindex: 2
    ifname: Ethernet2
    ipv4: 172.16.1.2/24
    node: s2
    vlan:
      access: blue
  - ifindex: 1
    ifname: eth1
    ipv4: 172.16.1.6/24
    node: h4
  linkindex: 4
  node_count: 2
  prefix:
    allocation: id_based
    ipv4: 172.16.1.0/24
  type: lan
- bridge: input_5
  gateway:
    ipv4: 172.16.2.2/24
  interfaces:
  - _vlan_mode: irb
    ifindex: 3
    ifname: Ethernet3
    ipv4: 172.16.2.2/24
    node: s2
    vlan:
      access: green
  - ifindex: 1
    ifname: eth1
    ipv4: 172.16.2.7/24
    node: h5
  linkindex: 5
  node_count: 2
  prefix:
    allocation: id_based
    ipv4: 172.16.2.0/24
  type: lan
- bridge: input_6
  gateway:
    ipv4: 172.16.2.2/24
  interfaces:
  - _vlan_mode: irb
    ifindex: 3
    ifname: GigabitEthernet0/3
    node: s1
    vlan:
      trunk:
        blue: {}
        red: {}
  - _vlan_mode: irb
    ifindex: 4
    ifname: Ethernet4
    ipv4: 172.16.2.2/24
    node: s2
    vlan:
      native: green
      trunk:
        blue: {}
        green: {}
        red: {}
  - ifindex: 2
    ifname: eth2
    ipv4: 172.16.2.3/24
    node: h1
  linkindex: 6
  node_count: 3
  prefix:
    allocation: id_based
    ipv4: 172.16.2.0/24
  type: lan
  vlan:
    trunk:
      blue: {}
      red: {}
- bridge: input_7
  interfaces:
  - _vlan_mode: irb
    ifindex: 4
    ifname: GigabitEthernet0/4
    ipv4: 172.16.2.1/24
    node: s1
    vlan:
      native: green
      trunk:
        blue: {}
        green: {}
        red: {}
  - _vlan_mode: irb
    ifindex: 5
    ifname: Ethernet5
    ipv4: 172.16.2.2/24
    node: s2
    vlan:
      native: green
      trunk:
        blue: {}
        green: {}
        red: {}
  linkindex: 7
  node_count: 2
  prefix:
    allocation: id_based
    ipv4: 172.16.2.0/24
  type: lan
  vlan:
    native: green
    trunk:
      blue: {}
      green: {}
      red: {}
module:
- vlan
- ospf
name: input
nodes:
  h1:
    af:
      ipv4: true
    box: generic/ubuntu2004
    device: linux
    id: 3
    interfaces:
    - bridge: input_1
      gateway:
        ipv4: 172.16.0.1/24
      ifindex: 1
      ifname: eth1
      ipv4: 172.16.0.3/24
      linkindex: 1
      name: h1 -> [s1,s2,h3]
      neighbors:
      - ifname: BVI1
        ipv4: 172.16.0.1/24
        node: s1
      - ifname: Vlan1000
        ipv4: 172.16.0.2/24
        node: s2
      - ifname: eth1
        ipv4: 172.16.0.5/24
        node: h3
      type: lan
    - bridge: input_6
      gateway:
        ipv4: 172.16.2.2/24
      ifindex: 2
      ifname: eth2
      ipv4: 172.16.2.3/24
      linkindex: 6
      name: h1 -> [s2,s1,h5]
      neighbors:
      - ifname: Vlan1002
        ipv4: 172.16.2.2/24
        node: s2
      - ifname: BVI3
        ipv4: 172.16.2.1/24
        node: s1
      - ifname: eth1
        ipv4: 172.16.2.7/24
        node: h5
      type: lan
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.103
      mac: 08:4f:a9:00:00:03
    name: h1
    role: host
  h2:
    af:
      ipv4: true
    box: generic/ubuntu2004
    device: linux
    id: 4
    interfaces:
    - bridge: input_2
      gateway:
        ipv4: 172.16.1.1/24
      ifindex: 1
      ifname: eth1
      ipv4: 172.16.1.4/24
      linkindex: 2
      name: h2 -> [s1,s2,h4]
      neighbors:
      - ifname: BVI2
        ipv4: 172.16.1.1/24
        node: s1
      - ifname: Vlan1001
        ipv4: 172.16.1.2/24
        node: s2
      - ifname: eth1
        ipv4: 172.16.1.6/24
        node: h4
      type: lan
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.104
      mac: 08:4f:a9:00:00:04
    name: h2
    role: host
  h3:
    af:
      ipv4: true
    box: generic/ubuntu2004
    device: linux
    id: 5
    interfaces:
    - bridge: input_3
      gateway:
        ipv4: 172.16.0.2/24
      ifindex: 1
      ifname: eth1
      ipv4: 172.16.0.5/24
      linkindex: 3
      name: h3 -> [h1,s1,s2]
      neighbors:
      - ifname: eth1
        ipv4: 172.16.0.3/24
        node: h1
      - ifname: BVI1
        ipv4: 172.16.0.1/24
        node: s1
      - ifname: Vlan1000
        ipv4: 172.16.0.2/24
        node: s2
      type: lan
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.105
      mac: 08:4f:a9:00:00:05
    name: h3
    role: host
  h4:
    af:
      ipv4: true
    box: generic/ubuntu2004
    device: linux
    id: 6
    interfaces:
    - bridge: input_4
      gateway:
        ipv4: 172.16.1.2/24
      ifindex: 1
      ifname: eth1
      ipv4: 172.16.1.6/24
      linkindex: 4
      name: h4 -> [h2,s1,s2]
      neighbors:
      - ifname: eth1
        ipv4: 172.16.1.4/24
        node: h2
      - ifname: BVI2
        ipv4: 172.16.1.1/24
        node: s1
      - ifname: Vlan1001
        ipv4: 172.16.1.2/24
        node: s2
      type: lan
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.106
      mac: 08:4f:a9:00:00:06
    name: h4
    role: host
  h5:
    af:
      ipv4: true
    box: generic/ubuntu2004
    device: linux
    id: 7
    interfaces:
    - bridge: input_5
      gateway:
        ipv4: 172.16.2.2/24
      ifindex: 1
      ifname: eth1
      ipv4: 172.16.2.7/24
      linkindex: 5
      name: h5 -> [s2,s1,h1]
      neighbors:
      - ifname: Vlan1002
        ipv4: 172.16.2.2/24
        node: s2
      - ifname: BVI3
        ipv4: 172.16.2.1/24
        node: s1
      - ifname: eth2
        ipv4: 172.16.2.3/24
        node: h1
      type: lan
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.107
      mac: 08:4f:a9:00:00:07
    name: h5
    role: host
  s1:
    af:
      ipv4: true
    box: cisco/iosv
    device: iosv
    id: 1
    interfaces:
    - bridge: input_1
      ifindex: 1
      ifname: GigabitEthernet0/1
      linkindex: 1
      type: lan
      vlan:
        access: red
        access_id: 1000
    - bridge: input_2
      ifindex: 2
      ifname: GigabitEthernet0/2
      linkindex: 2
      type: lan
      vlan:
        access: blue
        access_id: 1001
    - bridge: input_6
      ifindex: 3
      ifname: GigabitEthernet0/3
      linkindex: 6
      name: s1 -> [s2,h5,h1]
      neighbors:
      - ifname: Vlan1002
        ipv4: 172.16.2.2/24
        node: s2
      - ifname: eth1
        ipv4: 172.16.2.7/24
        node: h5
      - ifname: eth2
        ipv4: 172.16.2.3/24
        node: h1
      subif_index: 2
      type: lan
      vlan:
        mode: irb
    - bridge: input_7
      ifindex: 4
      ifname: GigabitEthernet0/4
      linkindex: 7
      subif_index: 2
      type: lan
      vlan:
        access_id: 1002
        native: green
        trunk:
          blue: {}
          green: {}
          red: {}
        trunk_id:
        - 1002
    - ifindex: 5
      ifname: GigabitEthernet0/3.1
      parent_ifindex: 3
      parent_ifname: GigabitEthernet0/3
      type: vlan_member
      virtual_interface: true
      vlan:
        access: blue
        access_id: 1001
    - ifindex: 6
      ifname: GigabitEthernet0/3.2
      parent_ifindex: 3
      parent_ifname: GigabitEthernet0/3
      type: vlan_member
      virtual_interface: true
      vlan:
        access: red
        access_id: 1000
    - ifindex: 7
      ifname: GigabitEthernet0/4.1
      parent_ifindex: 4
      parent_ifname: GigabitEthernet0/4
      type: vlan_member
      virtual_interface: true
      vlan:
        access: blue
        access_id: 1001
    - ifindex: 8
      ifname: GigabitEthernet0/4.2
      parent_ifindex: 4
      parent_ifname: GigabitEthernet0/4
      type: vlan_member
      virtual_interface: true
      vlan:
        access: red
        access_id: 1000
    - bridge_group: 1
      ifindex: 9
      ifname: BVI1
      ipv4: 172.16.0.1/24
      name: VLAN red (1000) -> [h1,s2,h3]
      neighbors:
      - ifname: eth1
        ipv4: 172.16.0.3/24
        node: h1
      - ifname: Vlan1000
        ipv4: 172.16.0.2/24
        node: s2
      - ifname: eth1
        ipv4: 172.16.0.5/24
        node: h3
      ospf:
        area: 0.0.0.0
        cost: 10
        passive: false
      type: svi
      virtual_interface: true
      vlan:
        mode: irb
        name: red
    - bridge_group: 2
      ifindex: 10
      ifname: BVI2
      ipv4: 172.16.1.1/24
      name: VLAN blue (1001) -> [h2,s2,h4]
      neighbors:
      - ifname: eth1
        ipv4: 172.16.1.4/24
        node: h2
      - ifname: Vlan1001
        ipv4: 172.16.1.2/24
        node: s2
      - ifname: eth1
        ipv4: 172.16.1.6/24
        node: h4
      ospf:
        area: 0.0.0.0
        passive: false
      type: svi
      virtual_interface: true
      vlan:
        mode: irb
        name: blue
    - bridge_group: 3
      ifindex: 11
      ifname: BVI3
      ipv4: 172.16.2.1/24
      name: VLAN green (1002) -> [s2,h5,h1]
      neighbors:
      - ifname: Vlan1002
        ipv4: 172.16.2.2/24
        node: s2
      - ifname: eth1
        ipv4: 172.16.2.7/24
        node: h5
      - ifname: eth2
        ipv4: 172.16.2.3/24
        node: h1
      ospf:
        area: 0.0.0.0
        passive: false
      type: svi
      virtual_interface: true
      vlan:
        mode: irb
        name: green
    loopback:
      ifindex: 0
      ifname: Loopback0
      ipv4: 10.0.0.1/32
      neighbors: []
      ospf:
        area: 0.0.0.0
      type: loopback
      virtual_interface: true
    mgmt:
      ifname: GigabitEthernet0/0
      ipv4: 192.168.121.101
      mac: 08:4f:a9:00:00:01
    module:
    - vlan
    - ospf
    name: s1
    ospf:
      af:
        ipv4: true
      area: 0.0.0.0
      router_id: 10.0.0.1
    vlan:
      max_bridge_group: 3
    vlans:
      blue:
        bridge_group: 2
        id: 1001
        mode: irb
        prefix:
          allocation: id_based
          ipv4: 172.16.1.0/24
      green:
        bridge_group: 3
        id: 1002
        mode: irb
        prefix:
          allocation: id_based
          ipv4: 172.16.2.0/24
      red:
        bridge_group: 1
        id: 1000
        mode: irb
        ospf:
          cost: 10
        prefix:
          allocation: id_based
          ipv4: 172.16.0.0/24
  s2:
    af:
      ipv4: true
    box: arista/veos
    device: eos
    id: 2
    interfaces:
    - bridge: input_3
      ifindex: 1
      ifname: Ethernet1
      linkindex: 3
      type: lan
      vlan:
        access: red
        access_id: 1000
    - bridge: input_4
      ifindex: 2
      ifname: Ethernet2
      linkindex: 4
      type: lan
      vlan:
        access: blue
        access_id: 1001
    - bridge: input_5
      ifindex: 3
      ifname: Ethernet3
      linkindex: 5
      type: lan
      vlan:
        access: green
        access_id: 1002
    - bridge: input_6
      ifindex: 4
      ifname: Ethernet4
      linkindex: 6
      type: lan
      vlan:
        access_id: 1002
        native: green
        trunk:
          blue: {}
          green: {}
          red: {}
        trunk_id:
        - 1000
        - 1001
        - 1002
    - bridge: input_7
      ifindex: 5
      ifname: Ethernet5
      linkindex: 7
      type: lan
      vlan:
        access_id: 1002
        native: green
        trunk:
          blue: {}
          green: {}
          red: {}
        trunk_id:
        - 1000
        - 1001
        - 1002
    - bridge_group: 1
      ifindex: 10
      ifname: Vlan1000
      ipv4: 172.16.0.2/24
      name: VLAN red (1000) -> [h1,s1,h3]
      neighbors:
      - ifname: eth1
        ipv4: 172.16.0.3/24
        node: h1
      - ifname: BVI1
        ipv4: 172.16.0.1/24
        node: s1
      - ifname: eth1
        ipv4: 172.16.0.5/24
        node: h3
      ospf:
        area: 0.0.0.0
        cost: 10
        passive: false
      type: svi
      virtual_interface: true
      vlan:
        mode: irb
        name: red
    - bridge_group: 2
      ifindex: 11
      ifname: Vlan1001
      ipv4: 172.16.1.2/24
      name: VLAN blue (1001) -> [h2,s1,h4]
      neighbors:
      - ifname: eth1
        ipv4: 172.16.1.4/24
        node: h2
      - ifname: BVI2
        ipv4: 172.16.1.1/24
        node: s1
      - ifname: eth1
        ipv4: 172.16.1.6/24
        node: h4
      ospf:
        area: 0.0.0.0
        passive: false
      type: svi
      virtual_interface: true
      vlan:
        mode: irb
        name: blue
    - bridge_group: 3
      ifindex: 12
      ifname: Vlan1002
      ipv4: 172.16.2.2/24
      name: VLAN green (1002) -> [s1,h5,h1]
      neighbors:
      - ifname: BVI3
        ipv4: 172.16.2.1/24
        node: s1
      - ifname: eth1
        ipv4: 172.16.2.7/24
        node: h5
      - ifname: eth2
        ipv4: 172.16.2.3/24
        node: h1
      ospf:
        area: 0.0.0.0
        passive: false
      type: svi
      virtual_interface: true
      vlan:
        mode: irb
        name: green
    loopback:
      ifindex: 0
      ifname: Loopback0
      ipv4: 10.0.0.2/32
      neighbors: []
      ospf:
        area: 0.0.0.0
      type: loopback
      virtual_interface: true
    mgmt:
      ifname: Management1
      ipv4: 192.168.121.102
      mac: 08:4f:a9:00:00:02
    module:
    - vlan
    - ospf
    name: s2
    ospf:
      af:
        ipv4: true
      area: 0.0.0.0
      router_id: 10.0.0.2
    vlan:
      max_bridge_group: 3
    vlans:
      blue:
        bridge_group: 2
        id: 1001
        mode: irb
        prefix:
          allocation: id_based
          ipv4: 172.16.1.0/24
      green:
        bridge_group: 3
        id: 1002
        mode: irb
        prefix:
          allocation: id_based
          ipv4: 172.16.2.0/24
      red:
        bridge_group: 1
        id: 1000
        mode: irb
        ospf:
          cost: 10
        prefix:
          allocation: id_based
          ipv4: 172.16.0.0/24
ospf:
  area: 0.0.0.0
provider: libvirt
vlans:
  blue:
    host_count: 2
    id: 1001
    neighbors:
    - ifname: eth1
      ipv4: 172.16.1.4/24
      node: h2
    - ifname: BVI2
      ipv4: 172.16.1.1/24
      node: s1
    - ifname: Vlan1001
      ipv4: 172.16.1.2/24
      node: s2
    - ifname: eth1
      ipv4: 172.16.1.6/24
      node: h4
    prefix:
      allocation: id_based
      ipv4: 172.16.1.0/24
  green:
    host_count: 2
    id: 1002
    neighbors:
    - ifname: Vlan1002
      ipv4: 172.16.2.2/24
      node: s2
    - ifname: BVI3
      ipv4: 172.16.2.1/24
      node: s1
    - ifname: eth1
      ipv4: 172.16.2.7/24
      node: h5
    - ifname: eth2
      ipv4: 172.16.2.3/24
      node: h1
    prefix:
      allocation: id_based
      ipv4: 172.16.2.0/24
  red:
    host_count: 2
    id: 1000
    neighbors:
    - ifname: eth1
      ipv4: 172.16.0.3/24
      node: h1
    - ifname: BVI1
      ipv4: 172.16.0.1/24
      node: s1
    - ifname: Vlan1000
      ipv4: 172.16.0.2/24
      node: s2
    - ifname: eth1
      ipv4: 172.16.0.5/24
      node: h3
    ospf:
      cost: 10
    prefix:
      allocation: id_based
      ipv4: 172.16.0.0/24
